### ISSUE TITLE
vulnerabilities are reported on versions not exist in NVD

### ISSUE THREAD

**Issue by biancajiang** [dependency-check/DependencyCheck#496]
Scanning apache tomcat v7.0.69 (http://archive.apache.org/dist/tomcat/tomcat-7/v7.0.69/) reports vulnerabilities only apply to "7.x before 7.0.68", eg:
https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-0714

Scanning tomcat v7.0.65 properly reports these vulnerabilities with high confidence which is great.

In fact, searching for apache tomcat on NVD shows the latest version for 7.0.6x existing is "7.0.67":
https://web.nvd.nist.gov/view/vuln/search-advanced?adv_search=true&cves=on&cpe_version=cpe%3a%2fa%3aapache%3atomcat

If such versions that do not exist on NVD can be filtered out, it would reduce quite a lot of false positives.

**Comment by jeremylong** [comment]
I'm somewhat confused by your post. I just scanned:

``` xml
<dependency>
    <groupId>org.apache.tomcat.embed</groupId>
    <artifactId>tomcat-embed-core</artifactId>
    <version>7.0.69</version>
</dependency>
```

Which is identified as cpe:/a:apache:tomcat:7.0.69 and CVE-2016-0714 was (correctly) not reported. What are you indicating as a reported false positive?

**Comment by biancajiang** [comment]
If I download tomcat-7.0.69.zip from:
http://search.maven.org/#artifactdetails%7Corg.apache.tomcat%7Ctomcat%7C7.0.69%7Cpom
I can reproduce it.  

But if I scan tomcat7-websocket-7.0.69.jar from:

```
    <dependency>
        <groupId>org.apache.tomcat</groupId>
        <artifactId>tomcat7-websocket</artifactId>
        <version>7.0.69</version>
    </dependency>
```

I don't see the same vulnerability reported.

Maybe I'm missing something?

[owasp-tomcat7069-vulnerability.html.zip](https://github.com/jeremylong/DependencyCheck/files/257063/owasp-tomcat7069-vulnerability.html.zip)

There are total of 148 vulnerabilities found from 10 dependencies (total of 31) from tomcat-7.0.69.zip. report attached.

**Comment by jeremylong** [comment]
I finally figured out why this is a bug; still trying to figure out the best way to resolve it. Basically, the version number identification algorithm fails in this case because we are using a newer version of the product and lower confidence version information is being used.

**Comment by aikebah** [comment]
@jeremylong Was looking if there were some old issues that I could pick up. But if I try running the current CLI against the zip-download from Maven Central. I do not see this false positive appearing. Was the bug you spotted fixed, but never reported on this issue, or is the bug still lurking inside DC and is it only hidden for this specific FP in the current release of DependencyCheck?

**Comment by v6ak** [comment]
I've seen this or a similar problem:

    <dependency>
        <groupId>io.netty</groupId>
        <artifactId>netty-all</artifactId>
        <version>4.1.0.Final</version>
    </dependency>

ODC 3.1.0 reports cpe:/a:netty_project:netty:4.1.0:**beta1**.

**Comment by v6ak** [comment]
In the case of Netty 4.1.0.Final and ODC 3.1.2, I don't see even the “beta” information to be anywhere in the evidence, it is just in the vulnerability description.

There is the pom.xml and a brief analysis of the scan result: https://gist.github.com/v6ak/4be1a1702d2f22ed8afa785fd93ca5eb

**Comment by jeremylong** [comment]
This should be resolved in 5.0.0.
